wasync/src/main/java/org/atmosphere/wasync/transport/WebSocketTransport.java

```diff
@@ -418,7 +427,19 @@ public void onError(Throwable t) {
             logger.trace("onError for {}", t);
             status = Socket.STATUS.ERROR;
             logger.debug("", t);
-            onFailure(t);
+
+            // On Android, ErrnoException is fired if lose connection (WIFI)
+            if(t.getClass().getName().equals("android.system.ErrnoException")){
+                if (options.reconnect()) {
+                    close(); // force release resources and reconnect
+                    tryReconnect();
+                }else{
+                    onFailure(new IOException(t.getMessage(), t));
+                }
+            }else{
+                onFailure(t);
+            }
+
```

Comment:
Thank you for your contribution! Could you please reformat this code according to everything else is formatted? Thanks!